app/models/subnet.rb

```diff
@@ -23,7 +23,9 @@ class Subnet < ActiveRecord::Base
 
   # This casts Subnet to Subnet::Ipv4 if no type is set
   def self.new(*attributes, &block)
-    return Subnet::Ipv4.new_without_cast(*attributes, &block) if self == Subnet
+    type = nil
```

Comment:
I think, we need it when `(h = attributes.first).is_a?(Hash)` is false, don't we?